notebooks/Gradient Boosted Regression Trees - Single Target.ipynb

```diff
@@ -0,0 +1,646 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "y_pred = clf_Ca.predict(X_test)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "skl.metrics.mean_squared_error(y_test,y_pred)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we find the mean squared error again, this time with only our selected features"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "y_important_pred = clf_Ca_important.predict(X_important_test)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "skl.metrics.mean_squared_error(y_test,y_important_pred)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Reducing the number of features shouldn't increase the error much - error may even go down!  If error increased by an uncomfortable amount, try decreasing the feature_importance threshold to include more features.\n",
+    "\n",
+    "At this point one would usually use Randomized or Grid Search cross-validation \n",
+    "to tune hyperparameters that optimize model performance.  This can be time-consuming,\n",
+    "so some pre-determined hyperparameters are provided in the cell below\n",
+    "if you're eager to immediately fit and examine a model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "params = {'n_estimators':3000,\n",
+    "          'max_depth':3,\n",
+    "          'min_samples_split':15,\n",
+    "          'min_samples_leaf':3,\n",
+    "          'max_features':0.8,\n",
+    "          'learning_rate':0.01}\n",
+    "clf_Ca_final = skl.ensemble.GradientBoostingRegressor(**params)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "clf_Ca_final.fit(X_important_train,y_train.values.ravel())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Take a look at our R-squared value on our test data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "clf_Ca_final.score(X_important_test,y_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's compute the deviance at each stage (n_estimators total stages) for the training and test sets.  Each stage does appear to improve performance!  This seems true with most of our targets, with the exception of Phosphorus (P), which as of this writing has been difficult to fit with trees."
```

Comment:
hmm yes, i think in the high-level description I'll mention how RandomForest works by making trees in parallel, while in contrast GBR is a more sequential process with a new tree each stage